WeakAuras/GenericTrigger.lua

```diff
@@ -3373,14 +3373,49 @@ function Private.ExecEnv.CheckTotemName(totemName, triggerTotemName, triggerTote
   return true
 end
 
-function WeakAuras.GetSpellCost(powerTypeToCheck)
-  local spellID = select(9, WeakAuras.UnitCastingInfo("player"))
-  if spellID then
-    local costTable = GetSpellPowerCost(spellID);
-    for _, costInfo in pairs(costTable) do
-      -- When there is no required aura for a power cost, the API returns an aura ID of 0 and false for hasRequiredAura despite being valid.
-      if costInfo.type == powerTypeToCheck and (costInfo.requiredAuraID == 0 or costInfo.hasRequiredAura) then
-        return costInfo.cost;
+do
+  local queuableSpells
+  if WeakAuras.IsClassicEraOrWrath() then
+    local classQueueableSpells = {
+      ["WARRIOR"] = {
+        78,    -- Heroic Strike
+        845,   -- Cleave
+      },
+      ["HUNTER"] = {
+        2973,  -- Raptor Strike
+      },
+      ["DRUID"] = {
+        6807,  -- Maul
+      },
+      ["DEATHKNIGHT"] = {
+        56815, -- Rune Strike
+      },
+    }
```

Comment:
A hard-coded table doesn't feel great, but the only alternative I could think of was looping through the spellbook and scanning tooltips for the Global String `SPELL_ON_NEXT_SWING = "Next melee"`.